src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibleBuffer.ts

```diff
@@ -0,0 +1,201 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { KeyCode } from 'vs/base/common/keyCodes';
+import { DisposableStore, IDisposable } from 'vs/base/common/lifecycle';
+import { URI } from 'vs/base/common/uri';
+import { IEditorConstructionOptions } from 'vs/editor/browser/config/editorConfiguration';
+import { EditorExtensionsRegistry } from 'vs/editor/browser/editorExtensions';
+import { CodeEditorWidget, ICodeEditorWidgetOptions } from 'vs/editor/browser/widget/codeEditorWidget';
+import { StringBuilder } from 'vs/editor/common/core/stringBuilder';
+import { ITextModel } from 'vs/editor/common/model';
+import { IModelService } from 'vs/editor/common/services/model';
+import { LinkDetector } from 'vs/editor/contrib/links/browser/links';
+import { localize } from 'vs/nls';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { TerminalCapability } from 'vs/platform/terminal/common/capabilities/capabilities';
+import { TerminalSettingId } from 'vs/platform/terminal/common/terminal';
+import { SelectionClipboardContributionID } from 'vs/workbench/contrib/codeEditor/browser/selectionClipboard';
+import { getSimpleEditorOptions } from 'vs/workbench/contrib/codeEditor/browser/simpleEditorOptions';
+import { ITerminalContribution, ITerminalInstance, IXtermTerminal } from 'vs/workbench/contrib/terminal/browser/terminal';
+import { TerminalWidgetManager } from 'vs/workbench/contrib/terminal/browser/widgets/widgetManager';
+import { XtermTerminal } from 'vs/workbench/contrib/terminal/browser/xterm/xtermTerminal';
+import { ITerminalFont, ITerminalProcessManager } from 'vs/workbench/contrib/terminal/common/terminal';
+import { Terminal } from 'xterm';
+
+const enum AccessibleBufferConstants {
+	Scheme = 'terminal-accessible-buffer'
+}
+
+export class AccessibleBufferWidget extends DisposableStore implements ITerminalContribution {
+	public static ID: string = AccessibleBufferConstants.Scheme;
+	private _accessibleBuffer: HTMLElement | undefined;
+	private _bufferEditor: CodeEditorWidget | undefined;
+	private _editorContainer: HTMLElement | undefined;
+	private _commandFinishedDisposable: IDisposable | undefined;
+	private _refreshSelection: boolean = true;
+	private _registered: boolean = false;
+	private _lastContentLength: number = 0;
+	private _font: ITerminalFont | undefined;
+	private _terminal: XtermTerminal | undefined;
```

Comment:
You could also do something similar to links which has the contribution wrap the link manager, so an accessible buffer contribution creates a AccessibleBufferWidget. That way, you don't need `these | undefined` checks are xtermReady will have definitely fired.